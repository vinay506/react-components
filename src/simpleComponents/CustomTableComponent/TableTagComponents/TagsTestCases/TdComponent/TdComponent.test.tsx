import React from 'react';
import { shallow } from 'enzyme';
import { CustomTdCell } from '../../index';
import renderer from 'react-test-renderer'


describe('<Custom TD Cell >', function () {
    let wrapper: any;
    let props: any;
    let tree: any;
    beforeEach(() => {
        props = {
            style:{fontWeight:'bold',height:'45px'}
        };
        wrapper = shallow(<CustomTdCell {...props}>Sample value for td</CustomTdCell>)
    });

    // Snapshot test
    it('should render', () => {
        tree = renderer.create(<CustomTdCell {...props}></CustomTdCell>).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('should render Td cell', () => {
        const td = wrapper.find('td');
        expect(td.exists()).toBe(true);
    })

    it('should match the value', () => {
        const value = wrapper.find('td').text();
        expect(value).toEqual('Sample value for td')
    })

    it('should match with classnames of td', () => {
        const props = wrapper.find('td').props();
        expect(props.className).toEqual('mdc-data-table__cell rmwc-data-table__cell')
    })

    it('should match with styles from props', () => {
        const props = wrapper.find('td').props();
        expect(props.style).toEqual({"fontWeight": "bold", "height": "45px"});
    })

    it('should display X on Empty', () => {
        const emptyCell = shallow(<CustomTdCell {...props}></CustomTdCell>)
        const text = emptyCell.find('td').text();
        expect(text).toEqual('X');
    })
    
})