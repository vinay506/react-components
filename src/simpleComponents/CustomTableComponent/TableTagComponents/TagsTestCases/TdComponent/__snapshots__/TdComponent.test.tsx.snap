// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Custom TD Cell > should render 1`] = `
<td
  className="mdc-data-table__cell rmwc-data-table__cell"
  style={
    Object {
      "fontWeight": "bold",
      "height": "45px",
    }
  }
>
  X
</td>
`;
